torch/distributions/transforms.py

```diff
@@ -0,0 +1,231 @@
+
+
+class ExpTransform(Transform):
+    """
+    Transform via the mapping `y = exp(x)`.
+    """
+    domain = constraints.real
+    codomain = constraints.positive
+
+    def _forward(self, x):
+        return x.exp()
+
+    def _inverse(self, y):
+        return y.log()
+
+    def log_abs_det_jacobian(self, x, y):
+        return x
+
+
+class SigmoidTransform(Transform):
+    """
+    Transform via the mapping `y = sigmoid(x)` and `x = logit(y)`.
+    """
+    domain = constraints.real
+    codomain = constraints.unit_interval
+
+    def _forward(self, x):
+        return sigmoid(x)
+
+    def _inverse(self, y):
+        return y.log() - (-y).log1p()
+
+    def log_abs_det_jacobian(self, x, y):
+        return -(y.reciprocal() + (1 - y).reciprocal()).log()
+
+
+class AbsTransform(Transform):
+    """
+    Transform via the mapping `y = abs(x)`
+    """
+    domain = constraints.real
+    codomain = constraints.positive
+
+    def _forward(self, x):
+        return x.abs()
+
+
+class AffineTransform(Transform):
+    """
+    Transform via the pointwise affine mapping `y = loc + scale * x`.
+
+    Args:
+        loc (Tensor or Variable): Location parameter.
+        scale (Tensor or Variable): Scale parameter.
+        event_dim (int): Optional size of `event_shape`. This should be zero
+            for univariate random variables, 1 for distributions over vectors,
+            2 for distributions over matrices, etc.
+    """
+    domain = constraints.real
+    codomain = constraints.real
+
+    def __init__(self, loc, scale, event_dim=0):
+        super(AffineTransform, self).__init__()
+        self.loc = loc
+        self.scale = scale
+        self.event_dim = event_dim
+
+    def _forward(self, x):
+        return self.loc + self.scale * x
+
+    def _inverse(self, y):
+        return (y - self.loc) / self.scale
+
+    def log_abs_det_jacobian(self, x, y):
+        result = torch.abs(self.scale).log()
+        shape = x.shape
+        for _ in range(self.event_dim):
+            result = result.sum(-1)
+            shape = shape[:-1]
+        return result.expand(shape)
+
+
+class LogprobTransform(Transform):
+    """
+    Transform from the simplex to unconstrained space via `y = log(x)`.
+
+    This is not bijective and cannot be used for HMC. However this acts mostly
+    coordinate-wise (except for the final normalization), and this is
+    appropriate for coordinate-wise optimization algorithms.
+    """
+    domain = constraints.simplex
+    codomain = constraints.positive
+
+    def _forward(self, x):
+        probs = x
+        return probs.log()
+
+    def _inverse(self, y):
+        logprobs = y
+        probs = (logprobs - logprobs.max(-1, True)[0]).exp()
+        probs /= probs.sum(-1, True)
+        return probs
+
+
+class StickBreakingTransform(Transform):
+    """
+    Transform from the simplex to unconstrained of one fewer dimension via a
+    stick-breaking process.

```

Comment:
Some explanation of stick-breaking (possibly a link) would be nice (it's not trivial like all other transforms)

Reply:
Done.